extensions/amp-instagram/0.1/amp-instagram.js

```diff
@@ -76,16 +82,43 @@ class AmpInstagram extends AMP.BaseElement {
   }
 
   /** @override */
-  prerenderAllowed() {
-    return true;
+  prerenderCallback() {
+    const image = new Image();
```

Comment:
So few problems with doing this inside `buildCallback`, namely AMP will build all elements regardless where they are inside the document. So all instagram images for example will be loaded on build which I thought we wanted to avoid to save data  - no need to unnecessarily load (during build) resources for articles the user might not even open.

If you're talking about just building the DOM of the image but not setting src or loading the image in build then that's a good suggestion yes.

As for using the placeholder to handle this, during review that was problemitic namely because placeholders have strict CSS rules with `!important` on them so one couldn't update the TRBL values, to do that we need to create some more DOM structure to handle that - which is possible and I am actually with.